src/DFG.cpp

```diff
@@ -133,28 +191,123 @@ void DFG::combineCmpBranch() {
   }
 }
 
-void DFG::combineMulAdd() {
+// Combine phi + iadd or phi + iadd + iadd where iadd is integer addition.
```

Comment:
When a `phi` has multiple `iadd`, I believe it would simply pick the first one no matter whether the second one has grandchild `iadd`, i.e., we may unfortunately skip the best opportunities of  maximum fusion, right? Plz add comment/example for this case. (no need to fix it as it might take some time.)

Reply:
u are right. I have added comments here.